WalletConnectSharp.Core/WalletConnectSession.cs

```diff
@@ -580,33 +580,34 @@ protected void HandleSessionDisconnect(string msg, string topic = "disconnect",
 
         if (TransportConnected)
         {
-            DisconnectTransport();
+            // TODO: this is a hack to get the transport to disconnect syncronously.
+            Task.WhenAll(DisconnectTransport());
```

Comment:
Realistically should make all the code that depends on this able to `await` this call.